docs/content/getting-started/production-best-practices.mdx

```diff
@@ -21,17 +21,53 @@ The following list outlines some guidelines and best practices for running OpenF
+
+The following table enumerates the server's concurrency specific settings:
+
+| flag                                    | env                                           | config                           |
+|-----------------------------------------|-----------------------------------------------|----------------------------------|
+| --max-concurrent-reads-for-list-objects | OPENFGA_MAX_CONCURRENT_READS_FOR_LIST_OBJECTS | maxConcurrentReadsForListObjects |
+| --max-concurrent-reads-for-check        | OPENFGA_MAX_CONCURRENT_READS_FOR_CHECK        | maxConcurrentReadsForCheck       |
+| --resolve-node-limit                    | OPENFGA_RESOLVE_NODE_LIMIT                    | resolveNodeLimit                 |
+| --resolve-node-breadth-limit            | OPENFGA_RESOLVE_NODE_BREADTH_LIMIT            | resolveNodeBreadthLimit          |
+
+
+Determining the right values for these settings will be based on a variety of factors including, but not limited to, the database specific deployment topology, the FGA model(s) involved, and the relationship tuples in the system. However, here are some high-level guidelines:
+
+* If a single ListObjects query is negatively impacting other query endpoints, then consider setting a stricter value for `OPENFGA_MAX_CONCURRENT_READS_FOR_LIST_OBJECTS`.
+
+* If a single Check query is negatively impacting other query endpoints, then consider setting a stricter value for `OPENFGA_MAX_CONCURRENT_READS_FOR_CHECK`.
+
+If you still see high request latencies despite the guidance above, then you may additionally consider setting stricter limits on the query resolution behavior by limiting the resolution depth and resolution breadth. These can be controlled with the `OPENFGA_RESOLVE_NODE_LIMIT` and `OPENFGA_RESOLVE_NODE_BREADTH_LIMIT` settings, respectively. Consider these guidelines:
+
+* `OPENFGA_RESOLVE_NODE_LIMIT` limits the resolution depth of a single query, and thus it sets an upper bound on how deep a relationship hierarchy may be. A high value will allow a single query to involve more hierarchical resolution and therefore more database queries, while a low value will reduce the number of hierarchical resolutions that will be allowed and thus reduce the number of database queries.
+
+* `OPENFGA_RESOLVE_NODE_BREADTH_LIMIT` limits the resolution breadth. It sets an upper bound on the number of in-flight resolutions that can be taking place on one or more [usersets](../concepts#what-is-a-user). A high value will allow a single query to involve more concurrent evaluations to take place and therefore more database queries and server processes, while a low value will reduce the overall number of concurrent resolutions that will be allowed and thus reduce the number of database queries and server processes.
```

Comment:
> `OPENFGA_RESOLVE_NODE_BREADTH_LIMIT` limits the resolution breadth.

Can give an example here, e.g. if a model has `define viewer: a1 or a2 or a3 or .... aN`, this setting puts a limit on how many of the `a`s can be resolved concurrently.

> A high value will allow a single query to involve more concurrent evaluations to take place **and therefore more database queries**

No, this setting doesn't control how many database reads are made. I would say

> A high value will allow a single query to involve more concurrent evaluations to take place and therefore more threads created

Reply:
> No, this setting doesn't control how many database reads are made. I would say

The more tuples we encounter in the iterator the more goroutines and database queries. They are definitely related. Each evaluation of a tuple we visit causes another branch of evaluation that causes both goroutines and database queries.